cdap-ui/app/cdap/components/ServiceStatus/index.js

```diff
@@ -14,47 +14,249 @@
+      } else if(this.props.status === 'OK'){
+        circle = <div className={classNames({"status-circle-green" : !this.props.isLoading, "status-circle-grey" : this.props.isLoading})}>
+                  {this.state.provisioned}
+                 </div>;
+      } else {
+        circle = <div className={classNames({"status-circle-red" : !this.props.isLoading, "status-circle-grey" : this.props.isLoading})} />;
+      }
+    } else {
+      circle = <div className="status-circle-grey" />;
+    }
+
+    let logUrl = this.MyDataSrc.constructUrl({
+      _cdapPath : `/system/services/${this.props.name}/logs/next?&maxSize=50`
+    });
+
+    let provisionBtnClasses = classNames('btn btn-default btn-primary set-provision-btn', {'provision-btn-with-warning' : this.state.serviceWarning});
+
+    return (
+      <div
+        onClick={this.toggleDropdown}
+        className="service-status"
+        onMouseOver={this.toggleErrorMessage}
+        onMouseOut={this.toggleErrorMessage}
+      >
+        {circle}
+        <div className="status-label">
+          {T.translate(`features.Management.Services.${this.props.name.split('.').join('_')}`)}
+          {
+            this.state.serviceWarning ?
+              <span
+                className="fa fa-exclamation-triangle"
+                aria-hidden="true"
+              >
+              {
+                this.state.showError && !this.state.isDropdownOpen?
+                <div className="service-error-message">
+                  {this.state.errorText}
+                </div>
+                :
+                null
+              }
+              </span>
+              :
+              null
+          }
+        </div>
+        <div className="service-dropdown-container pull-right">
+          <Dropdown
+            isOpen={this.state.isDropdownOpen}
+            toggle={this.toggleDropdown}
+            className="service-dropdown"
+          >
+            <span className="fa fa-caret-down service-dropdown-caret">
+            </span>
+            <DropdownMenu>
+              <div className="dropdown-service-name service-dropdown-item">
+                {circle}
+                <div className="status-label">
+                  {T.translate(`features.Management.Services.${this.props.name.split('.').join('_')}`)}
+                </div>
+              </div>
+              <a href={logUrl} target="_blank">
+                <div className="service-dropdown-item">
+                  <span className="fa fa-file-text" />
+                    View Logs
+                </div>
+              </a>
+              <div
+                className="provision-dropdown-item service-dropdown-item"
+                onClick={this.provisionRowClickHandler}
+                onMouseOver={this.toggleErrorMessage}
+                onMouseOut={this.toggleErrorMessage}
+              >
+                Requested
+                <input
+                  className="provision-input"
+                  autoFocus
```

Comment:
`autoFocus` here won't be useful as there is a time delay in the dropdown getting rendered and focusing on the textbox. This has to be handled in a different way.